### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by darkdragon-001**
I collected a few improvements for things I had to fix before I was able to run the tests.

Unfortunately with an Ubuntu 24.04 host and container, only tests using `--privileged` succeed, the ones without are all failing now.

Edit: I noticed that I could get rid of some errors when *removing* the `SYS_ADMIN` capability.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ pip install -r requirements.txt
 
 Then run the tests with:
 ```sh
-pytest --log-cli-level=debug
+python3 -m pytest --log-cli-level=debug
```
Comment (LewisGaul) [inline README.md:104]: This shouldn't be necessary

File: tests/conftest.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ def pytest_configure(config: pytest.Config) -> None:
     # Set 'cgroup_version' config value.
     try:
         output = ctr_client.run(
-            "ubuntu:20.04",
+            "ubuntu:24.04",
```
Comment (LewisGaul) [inline tests/conftest.py:92]: These tests were deliberately written to test systemd in ubuntu:20.04. I'm not currently using the tests, but wouldn't want to remove support for testing 20.04 right now, so would only accept support for 22.04 and 24.04 if there's a clean way to support multiple at once (perhaps chosen via a CLI arg).

File: tests/conftest.py [hunk 3]
```diff
@@ -361,7 +361,7 @@ def default_ctr_kwargs(ctr_mgr: CtrMgr, setup_mode: Optional[str]) -> dict[str,
         kwargs["systemd"] = "always"
     else:
         # Systemd expects /run to be a tmpfs.
-        kwargs["tmpfs"] = ["/run"]
+        kwargs["tmpfs"] = ["/run", "/run/lock"]
```
Comment (LewisGaul) [inline tests/conftest.py:364]: Why is this needed? If `/run` is a tmpfs anyway then `/run/lock` is at least within a tmpfs. Is there systemd documentation stating this requirement?

File: tests/setup_modes/cgroupns/init_script.sh
Comment (LewisGaul) [inline tests/setup_modes/cgroupns/init_script.sh:1]: I'm not sure how this ever worked without these scripts being executable! Possibly because I was running on WSL from the Windows filesystem.

**Review by LewisGaul — COMMENTED** [review]
Thanks for the PR after trying things out. However, I'm afraid I'm not particularly keen on the change to a newer Ubuntu version right now (although I'm glad to see you didn't need a lot of changes to get that running!).

File: README.md (hunk 1 above)
Comment (darkdragon-001) [reply README.md:104]: Without this, it uses the system `pytest` version instead of the one from the virtualenv. If there is no system `pytest` installed, it fails.

File: tests/conftest.py (hunk 3 above)
Comment (darkdragon-001) [reply tests/conftest.py:364]: For some reason, it just doesn't seem to create the folder but just assumes it exists.

File: tests/setup_modes/cgroupns/init_script.sh
Comment (darkdragon-001) [reply tests/setup_modes/cgroupns/init_script.sh:1]: I was running on Ubuntu 24.04.

File: README.md (hunk 1 above)
Comment (LewisGaul) [reply README.md:104]: If you have the virtualenv active and pytest installed inside it then just invoking `pytest` should pick up that one.

File: tests/conftest.py (hunk 3 above)
Comment (LewisGaul) [reply tests/conftest.py:364]: Systemd might be trying to create a tmpfs mount at `/run/lock` if it doesn't exist, which would fail if there's no `CAP_SYS_ADMIN`. Is that the problem? I'm still not sure *why* systemd feels the need for this to exist.

Comment (darkdragon-001) [reply tests/conftest.py:364]: This could be the case. I don't know the reasoning, I just read at several comments that this is required and when adding it, some problems disappeared.

File: README.md (hunk 1 above)
Comment (darkdragon-001) [reply README.md:104]: For some reason, it didn't work for me.

File: tests/conftest.py (hunk 2 above)
Comment (darkdragon-001) [reply tests/conftest.py:92]: Ubuntu 20.04 will be EOL in 7 months, so I don't think it makes sense to waste resources for keeping this. At least I personally do not plan to spend time on this.

Comment (LewisGaul) [reply tests/conftest.py:92]: That's okay, I'm just letting you know that I'm not really open to accepting this change (and the associated changes such as requiring `/run/lock` that we don't fully understand) since my use-case hasn't currently moved on from ubuntu20.04 (it's the systemd version that's more relevant than the distro version, so the ubuntu EOL doesn't particularly bother me).
